04/src/academy/pocu/comp2500samples/w04/singletonlogger/Logger.java

```diff
@@ -0,0 +1,130 @@
+package academy.pocu.comp2500samples.w04.singletonlogger;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.util.List;
+
+public class Logger {
+    private static final String CONFIG_FILENAME = "logger-config.txt";
+
+    private static Logger instance;
+
+    private LogLevel minLogLevel;
+    private String outputPath;
+    private BufferedWriter outBuffer;
+
+    private Logger(LogLevel minLogLevel, String outputPath) {
+        this.minLogLevel = minLogLevel;
+        this.outputPath = outputPath;
+
+        try {
+            this.outBuffer = new BufferedWriter(new FileWriter(this.outputPath));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static Logger getInstance() {
+        if (instance == null) {
+            try {
+                String classPath = getClassPath();
+                Path loggerConfigPath = Paths.get(classPath, CONFIG_FILENAME);
+
+                File configFile = new File(loggerConfigPath.toString());
+
+                LogLevel logLevel = LogLevel.WARNING;
+                String outputFilename = "log.txt";
+
+                if (configFile.isFile()) {
+                    List<String> lines = Files.readAllLines(loggerConfigPath, StandardCharsets.UTF_8);
+
+                    for (String line : lines) {
+                        String[] splits = line.split("=");
+
+                        switch (splits[0]) {
+                            case "loglevel":
+                                logLevel = LogLevel.valueOf(splits[1]);

```

Comment:
if text file is wrong this could throw null pointer exception i think. do we want to have proper handling of this? or just let null exception happen :)?

Reply:
let it happen lol. Don't want to put in too much code that's not really relevant to the concept we are trying to show. I'll just mention it in the code slides